src/Shared/AssemblyNameExtension.cs

```diff
@@ -993,5 +1004,237 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             info.AddValue("immutable", immutable);
             info.AddValue("remapped", remappedFrom);
         }
+
+#if !NET35
+        internal class Converter : JsonConverter<AssemblyNameExtension>
+        {
+            public override AssemblyNameExtension Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                AssemblyNameExtension ane = new AssemblyNameExtension();
+                if (reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new JsonException();
+                }
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonTokenType.EndObject)
+                    {
+                        return ane;
+                    }
+                    else if (reader.TokenType == JsonTokenType.Null)
+                    {
+                        return null;
+                    }
+                    else if (reader.TokenType != JsonTokenType.PropertyName)
+                    {
+                        throw new JsonException();
+                    }
+                    string parameter = reader.GetString();
+                    reader.Read();
```

Comment:
What is this capturing exactly? Could you comment the first use of reader.Read() in this context?

Reply:
This probably does deserve a comment, since it's kinda overloaded.

The Json reader is set up such that every thing (technical term) of interest has its own type. In particular, when specifying that a particular property has a particular value, it has the "PropertyName" token followed by the token(s) representing the property. If that property is not null, it will start with a "StartObject" token if it's an object or a "StartArray" token if it's an array, etc. If it's null, it just starts with null, and there isn't anything else.

In this case, I read the first token. If it's null, that means that the json has something like "propertyName"=null, which is valid though avoided unless you modify it by hand. Having the null check you commented on means that we don't keep reading the object if it happens to be null. GetString() would have returned a valid property name on the ANE that wasn't set.

If the property isn't null, it skips over the "StartObject", etc. and lets us start reading the interesting stuff immediately. Does that make sense?